mesonbuild/interpreter.py

```diff
@@ -2263,6 +2264,66 @@ def _get_native() -> str:
             else:
                 return _get_native()
 
+    @permittedKwargs({'encoding', 'allow_absolute'})
+    @FeatureNew('meson.read_file', '0.58.0')
+    def read_file_method(
+            self, args: T.Sequence[str], kwargs: T.Dict[str, T.Any]) -> str:
```

Comment:
args needs to be updated. iIRC there's a `FileorString` alias in mesonlib for this.